plugins/alerting/pkg/alerting/drivers/alerting_manager/cluster_driver.go

```diff
@@ -70,66 +48,57 @@ func NewAlertingManagerDriver(options AlertingManagerDriverOptions) (*AlertingMa
 		options.K8sClient = c
 	}
 
-	return &AlertingManager{
-		AlertingManagerDriverOptions: options,
+	return &AlertingClusterManager{
+		AlertingDriverOptions: options,
 	}, nil
 }
 
-func (a *AlertingManager) GetClusterConfiguration(ctx context.Context, _ *emptypb.Empty) (*alertops.ClusterConfiguration, error) {
-	existing := a.newOpniGateway()
-
-	err := a.K8sClient.Get(ctx, client.ObjectKeyFromObject(existing), existing)
-	if err != nil {
-		return nil, err
-	}
-
-	spec := extractGatewayAlertingSpec(existing)
-	return &alertops.ClusterConfiguration{
-		NumReplicas:             spec.Replicas,
-		ClusterSettleTimeout:    spec.ClusterSettleTimeout,
-		ClusterPushPullInterval: spec.ClusterPushPullInterval,
-		ClusterGossipInterval:   spec.ClusterGossipInterval,
-		ResourceLimits: &alertops.ResourceLimitSpec{
-			Cpu:     spec.CPU,
-			Memory:  spec.Memory,
-			Storage: spec.Storage,
+func (a *AlertingClusterManager) newAlertingClusterCrd() *corev1beta1.AlertingCluster {
+	return &corev1beta1.AlertingCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "opni-alerting",
+			Namespace: a.GatewayRef.Namespace,
+		},
+		Spec: corev1beta1.AlertingClusterSpec{
+			Gateway: k8scorev1.LocalObjectReference{
+				Name: a.GatewayRef.Name,
+			},
 		},
-	}, nil
-}
-
-func (a *AlertingManager) ConfigureCluster(ctx context.Context, conf *alertops.ClusterConfiguration) (*emptypb.Empty, error) {
-	if err := conf.Validate(); err != nil {
-		return nil, err
 	}
-	lg := a.Logger.With("action", "configure-cluster")
-	lg.Debugf("%v", conf)
-	existing := a.newOpniGateway()
+}
 
-	err := a.K8sClient.Get(ctx, client.ObjectKeyFromObject(existing), existing)
-	if err != nil {
-		return nil, err
-	}
-	lg.Debug("got existing gateway")
-	mutator := func(gateway *corev1beta1.Gateway) {
-		gateway.Spec.Alerting.Replicas = conf.NumReplicas
-		gateway.Spec.Alerting.ClusterGossipInterval = conf.ClusterGossipInterval
-		gateway.Spec.Alerting.ClusterPushPullInterval = conf.ClusterPushPullInterval
-		gateway.Spec.Alerting.ClusterSettleTimeout = conf.ClusterSettleTimeout
-		gateway.Spec.Alerting.Storage = conf.ResourceLimits.Storage
-		gateway.Spec.Alerting.CPU = conf.ResourceLimits.Cpu
-		gateway.Spec.Alerting.Memory = conf.ResourceLimits.Memory
+func (a *AlertingClusterManager) InstallCluster(ctx context.Context, empty *emptypb.Empty) (*emptypb.Empty, error) {
```

Comment:
I think it's probably worthwhile to separate the specific driver implementation (kubernetes and others) from the implementation of the gRPC interface.  @kralicky thoughts?

Reply:
To expand on this the driver should be something the implementation of the gRPC abstraction uses rather than being the entirety of the gRPC interface.  I think there's value in separating these two concepts.

Reply:
yeah agreed, this is part of the reason I added AlertOpsV2 spec to the epic this part of, to rework these issues : https://github.com/rancher/opni/issues/1153 . It is definitely not the best to couple the type contract for these definitions

Although the ops server is separated from the driver in a pretty thin encapsulation layer :

https://github.com/rancher/opni/blob/e48247bd6eaa78a2a2ffe96eff8d95ba5246cd6b/plugins/alerting/pkg/alerting/ops/ops.go

Reply:
Also since the epic is moving towards deploying cortex-alertmanager anyways, it might be more productive to rework the driver interface later when tackling https://github.com/rancher/opni/issues/1153

Reply:
yeah I agree @dbason, it would be best if the driver is part of the implementation of that server.